presto-main/src/main/java/com/facebook/presto/cost/ValuesStatsRule.java

```diff
@@ -87,12 +88,16 @@ public ValuesStatsRule(Metadata metadata)
 
     private SymbolStatsEstimate buildSymbolStatistics(List<Object> values, Session session, Type type)
     {
+        if (values.isEmpty()) {
+            return new SymbolStatsEstimate(NaN, NaN, NaN, NaN, 0.0);
+        }
+
         List<Object> nonNullValues = values.stream()
                 .filter(Objects::nonNull)
                 .collect(toImmutableList());
 
         if (nonNullValues.isEmpty()) {
-            return SymbolStatsEstimate.ZERO_STATS;
+            return new SymbolStatsEstimate(NaN, NaN, 1.0, 0.0, 0.0);
```

Comment:
average row size is NaN

Reply:
If all values are null - row size is zero. If row count is zero - row size is NaN